docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -0,0 +1,173 @@
+
+To obtain a browser alias, create a client function that uses the [navigator.userAgent](https://www.w3schools.com/jsref/prop_nav_useragent.asp) property. The function will return a user-agent header sent by the browser. To parse the returned header, you can use external modules, for example, [ua-parser-js](https://github.com/faisalman/ua-parser-js).
+
+The following sample demonstrates how to create conditional test logic based on the used browser. Before running this test, install *ua-parser-js* by running `npm install ua-parser-js`.
+
+```js
+import { ClientFunction } from 'testcafe';
+import uaParser from 'ua-parser-js';
+
+fixture `My fixture`
+    .page `http://example.com`;
+
+//Returns a user-agent header sent by the browser
+const getUA = ClientFunction(() => navigator.userAgent);
+
+test('My Test', async t => {
+    const ua = await getUA();
+    const browserAlias =uaParser(ua).browser.name;
+
+    //Some common actions
+
+    //Conditional logic
+    if (browserAlias === 'Chrome'){
+        //Test logic for Chrome
+        console.log('The browser is Chrome');
+    }
+    else if (browserAlias === 'Safari'){
+        //Test logic for Safari
+        console.log('The browser is Safari');
+    }
+});
+```
+
+## Injecting External Libraries into a Page from a Test
+
+One more case of using client functions is injecting external libraries like [jQuery](https://jquery.com/) into a tested page.
+
+In order to work with jQuery from a test, you first need to add jQuery code to the tested page by using the `eval` method.
+
+```js
+await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+```
+
+After that,  you can use jQuery in the test within a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+import fs from 'fs';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    // Adds jQuery code to the page
+    await t.eval(new Function(fs.readFileSync('./jquery.min.js').toString()));
+
+    const getElementText = ClientFunction(() => {
+        // Uses the jQuery selector
+        return $('div.column.col-2').find('legend').eq(0).text();
+    });
+
+    await t.expect(getElementText()).contains('What is your primary Operating System');
+});
+```
+
+## Accessing Child Nodes in the DOM Hierarchy
+
+To find all child elements (not [nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node)) of nodes in a DOM hierarchy, TestCafe provides the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, you may need to access child nodes of some node. You can do this by using a client function.
```

Comment:
This phrase is rather difficult to grasp because there are many `node` and `element` words and you need to track where we use one or the other. How about this?

```
For each DOM node, TestCafe allows you to search through its child **elements** using the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, if you need too include child **nodes** in your search results, use client functions.
```